tachyon/math/finite_fields/extension_field_base.h

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) 2022 The Plonky3 Authors
+// Use of this source code is governed by a MIT/Apache-2.0 style license that
+// can be found in the LICENSE-MIT.plonky3 and the LICENCE-APACHE.plonky3
+// file.
+
+#ifndef TACHYON_MATH_FINITE_FIELDS_EXTENSION_FIELD_BASE_H_
+#define TACHYON_MATH_FINITE_FIELDS_EXTENSION_FIELD_BASE_H_
+
+#include <vector>
+
+#include "tachyon/math/finite_fields/extended_packed_field_traits_forward.h"
+#include "tachyon/math/finite_fields/packed_field_traits_forward.h"
+
+namespace tachyon::math {
+
+template <typename Derived>
+class ExtensionFieldBase {
+ public:
+  template <typename Derived2 = Derived,
```

Comment:
Can we use T instead of Derived2?

Derived2 is the type of the input `base`, so there is no case where Derived2 is not given.

Reply:
Is there a case where Derived2 and Derived are different?

Reply:
This structure is used when using `enable_if_t` with a given template type of the class. `Derived2` and `Derived` should be the same. Running without this structure and directly with Derived results in the error `error: no type named 'type' in 'std::enable_if<false>'; 'enable_if' cannot be used to disable this declaration`